File: backend/src/main/java/com/example/isa/service/interfaces/BloodSupplyService.java
package com.example.isa.service.interfaces;

import com.example.isa.model.BloodSupply;

import java.util.List;
import java.util.UUID;

public interface BloodSupplyService {
    List<BloodSupply> getAll();

    BloodSupply getById(UUID id);

    BloodSupply update(BloodSupply supplies);
}

File: frontend/src/app/services/BloodSupplyService.ts
import { HttpHeaders } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { ApiService } from "./ApiService";
import { ConfigService } from "./ConfigService";
import { BloodSupply } from '../model/BloodSupply';

@Injectable({
  providedIn: 'root'
})
export class BloodSupplyService {

  headers: HttpHeaders = new HttpHeaders({ 'Content-Type': 'application/json' });

  constructor(
    private apiService: ApiService,
    private config: ConfigService) {
  }

  getAll(): Observable<BloodSupply[]> {
    return this.apiService.get(`${this.config.blood_supplies_url}`);
  }

  update(supplies: BloodSupply): Observable<void>{
    return this.apiService.put(`${this.config.blood_supplies_url}/update`, supplies);
  }

}
